src/core/main.qml

```diff
@@ -47,121 +47,152 @@ Window {
     width: ApplicationSettings.previousWidth
     height: ApplicationSettings.previousHeight
     minimumWidth: 400 * ApplicationInfo.ratio
-    minimumHeight: 400 * ApplicationInfo.ratio
+    minimumHeight: 400 * ApplicationInfo.ratio 
     title: "GCompris"
-
+    
+    
     /// @cond INTERNAL_DOCS
-
+    
     property var applicationState: Qt.application.state
 
+    property var selectedMusicIndex 
+    property var rccBackgroundMusic: ApplicationInfo.getBackgroundMusicFromRcc();
+    property alias backgroundMusic: backgroundMusic
+    property bool isBackgroundMusicAllowed: ApplicationSettings.isBackgroundMusicEnabled
     onApplicationStateChanged: {
         if (ApplicationInfo.isMobile && applicationState != Qt.ApplicationActive) {
             audioVoices.stop();
             audioEffects.stop();
         }
     }
-
+    
     onClosing: Core.quit(main)
+   
+    onIsBackgroundMusicAllowedChanged: {
+        if(isBackgroundMusicAllowed) {
+            selectedMusicIndex = Math.floor(Math.random()*rccBackgroundMusic.length);
+            backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/" + rccBackgroundMusic[selectedMusicIndex]));
+        }
+        else {
+            backgroundMusic.stop()
+        }
+    }
 
     GCAudio {
         id: audioVoices
         muted: !ApplicationSettings.isAudioVoicesEnabled
-
+        
         Timer {
             id: delayedWelcomeTimer
             interval: 10000 /* Make sure, that playing welcome.ogg if delayed
                              * because of not yet registered voices, will only
                              * happen max 10sec after startup */
             repeat: false
-
+            
             onTriggered: {
                 DownloadManager.voicesRegistered.disconnect(playWelcome);
             }
-
+            
             function playWelcome() {
                 audioVoices.append(ApplicationInfo.getAudioFilePath("voices-$CA/$LOCALE/misc/welcome.$CA"));
             }
         }
-
+        
         Component.onCompleted: {
-            if(ApplicationSettings.isAudioEffectsEnabled)
-                append(ApplicationInfo.getAudioFilePath("qrc:/gcompris/src/core/resource/intro.$CA"))
+            if(!ApplicationSettings.isAudioEffectsEnabled && isBackgroundMusicAllowed) {
+                 Math.floor(Math.random()*rccBackgroundMusic.length);
```

Comment:
No, the background music was played only when audio effects was enabled earlier but there can be a case when the user doesn't want to keep the audio effects but wants to play the music? So, I just kept it as a check if audio effects is disabled then also the user can play the music.